### ISSUE TITLE
code's running environment

### ISSUE THREAD

**Issue by CK020717** [HKUDS/UrbanGPT#17]
I want to know what kind of equipment is needed to run this code. I'm getting an error running with rtx2060。

**Comment by LZH-YS1998** [comment]
Hello. If you are running the code only for evaluation purposes, you may need more than 18GB of GPU memory. If you plan to train the model, it will require over 32GB of GPU memory.

**Comment by CK020717** [comment]
> Hello. If you are running the code only for evaluation purposes, you may need more than 18GB of GPU memory. If you plan to train the model, it will require over 32GB of GPU memory.你好。如果您仅出于评估目的运行代码，则可能需要超过 18GB 的 GPU 内存。如果您计划训练模型，则需要超过 32GB 的 GPU 内存。

If I use an A40 GPU with 48GB of VRAM and 80GB of memory, approximately how long will it take to train? What configuration did you use to train your model files, and approximately how long did it take?

**Comment by CK020717** [comment]
> Hello. If you are running the code only for evaluation purposes, you may need more than 18GB of GPU memory. If you plan to train the model, it will require over 32GB of GPU memory.你好。如果您仅出于评估目的运行代码，则可能需要超过 18GB 的 GPU 内存。如果您计划训练模型，则需要超过 32GB 的 GPU 内存。

Hello. If I use an A40 GPU with 48GB of VRAM and 80GB of memory, approximately how long will it take to train? What configuration did you use to train your model files, and approximately how long did it take?

**Comment by LZH-YS1998** [comment]
This can be found in '4.1.2 Hyperparameters Settings' in our camera-ready version. 

During the instruction-tuning phase, we maintained consistency with Vicuna’s default settings for the majority of hyperparameters. These include a learning rate of 0.002, using the AdamW optimizer, and a warmup rate of 0.03. To improve training efficiency, we set the batch size to 4. The parameters for the dilation convolution kernel in the time encoder are set as follows: 𝑑𝑖𝑛, 𝑑𝑜𝑢𝑡, and 𝑑 ′ 𝑜𝑢𝑡 are all set to 32, with a dilation factor of 1. For our prediction task, both the history length (𝐻) and prediction length (𝑃) are set to 12. The projection layer parameters are configured with 𝑑 set to 64 and 𝑑𝐿 set to 4096. Lastly, the hidden layer parameter 𝑑 ′ for the regression layer is set to 128. We generated a total of 863,040 training samples from the NYC-taxi, NYC-bike, and NYC-crime datasets for instruction-tuning. The training process was conducted on 8 × NVIDIA A100-SXM4-40GB GPUs and took approximately 16 hours to complete in 3 epochs.
